RecoTracker/TkSeedGenerator/plugins/JetCoreMCtruthSeedGenerator.h

```diff
@@ -0,0 +1,193 @@
+
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeedCollection.h"
+
+#include "SimDataFormats/TrackingHit/interface/PSimHit.h"
+
+#include "TTree.h"
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+
+namespace edm {
+  class Event;
+  class EventSetup;
+}  // namespace edm
+
+class JetCoreMCtruthSeedGenerator : public edm::one::EDProducer<edm::one::SharedResources> {
+public:
+  explicit JetCoreMCtruthSeedGenerator(const edm::ParameterSet&);
+  ~JetCoreMCtruthSeedGenerator() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  // A pointer to a cluster and a list of tracks on it
+  struct TrackAndState {
+    TrackAndState(const reco::Track* aTrack, TrajectoryStateOnSurface aState) : track(aTrack), state(aState) {}
+    const reco::Track* track;
+    TrajectoryStateOnSurface state;
+  };
+
+  template <typename Cluster>
+  struct ClusterWithTracks {
+    ClusterWithTracks(const Cluster& c) : cluster(&c) {}
+    const Cluster* cluster;
+    std::vector<TrackAndState> tracks;
+  };
+
+  typedef ClusterWithTracks<SiPixelCluster> SiPixelClusterWithTracks;
+
+  typedef boost::sub_range<std::vector<SiPixelClusterWithTracks>> SiPixelClustersWithTracks;
+
+  TFile* JetCoreMCtruthSeedGenerator_out;
+  TTree* JetCoreMCtruthSeedGeneratorTree;
+
+  double jet_pt;
+  double jet_eta;
+  double pitchX = 0.01;   //100 um (pixel pitch in X)
+  double pitchY = 0.015;  //150 um (pixel pitch in Y)
+  bool print = false;
+  bool inclusiveConeSeed =
+      true;  //true= fill tracks in a cone of deltaR_, false=fill tracks which have SimHit on globDet
+
+private:
+  void beginJob() override;
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  void endJob() override;
+
+  // ----------member data ---------------------------
+  std::string propagatorName_;
+  edm::ESHandle<MagneticField> magfield_;
+  edm::ESHandle<GlobalTrackingGeometry> geometry_;
+  edm::ESHandle<Propagator> propagator_;
+
+  edm::EDGetTokenT<std::vector<reco::Vertex>> vertices_;
+  edm::EDGetTokenT<edmNew::DetSetVector<SiPixelCluster>> pixelClusters_;
+  std::vector<SiPixelClusterWithTracks> allSiPixelClusters;
+  std::map<uint32_t, SiPixelClustersWithTracks> siPixelDetsWithClusters;
+  edm::Handle<edm::DetSetVector<PixelDigiSimLink>> pixeldigisimlink;
+  edm::Handle<edmNew::DetSetVector<SiPixelCluster>> inputPixelClusters;
+  edm::EDGetTokenT<edm::DetSetVector<PixelDigiSimLink>> pixeldigisimlinkToken;
+  edm::EDGetTokenT<edm::View<reco::Candidate>> cores_;
+  edm::EDGetTokenT<std::vector<SimTrack>> simtracksToken;
+  edm::EDGetTokenT<std::vector<SimVertex>> simvertexToken;
+  edm::EDGetTokenT<std::vector<PSimHit>> PSimHitToken;
+  edm::Handle<std::vector<PSimHit>> simhits;
+
+  double ptMin_;
+  double deltaR_;
+  double chargeFracMin_;
+  double centralMIPCharge_;
+
+  std::string pixelCPE_;
+
+  tensorflow::GraphDef* graph_;
```

Comment:
graph should be in a cache object (see comments above)

Reply:
removed